src/api/payment/services/payment.service.ts

```diff
@@ -1,21 +1,81 @@
-import { Injectable } from '@nestjs/common';
+import { Body, Inject, Injectable } from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
-import { Repository } from 'typeorm';
-import { CreatePaymentDTO } from '../models/payment.dto';
+import Stripe from 'stripe';
+import { DataSource, Repository } from 'typeorm';
+import { Booking, bookingStatus } from '../../booking/models/booking.entity';
+import { BookingService } from '../../booking/services/booking.service';
+import { CreateCheckoutSessionDTO } from '../models/payment.dto';
 import { Payment } from '../models/payment.entity';
+import { StripeService } from './stripe.service';
 
 @Injectable()
 export class PaymentService {
   @InjectRepository(Payment)
-  private readonly repository: Repository<Payment>;
+  private readonly paymentRepository: Repository<Payment>;
 
-  public getPayment(id: string): Promise<Payment> {
-    return this.repository.findOneBy({ id: id });
+  @InjectRepository(Booking)
+  private readonly bookingRepository: Repository<Booking>;
+
+  @Inject(BookingService)
+  private readonly bookingService: BookingService;
+
+  @Inject(StripeService)
+  private readonly stripeService: StripeService;
+
+  @Inject(DataSource)
+  private readonly dataSource: DataSource;
+
+  public async createCheckoutSession(body: CreateCheckoutSessionDTO) {
+    // Create booking in our DB
+    const booking = await this.bookingService.createBooking(body);
+
+    return await this.stripeService.createCheckoutSession(
+      body.amount,
+      'https://www.google.com',
+      'https://www.bing.com',
```

Comment:
Should you set the redirectUrl to localhost:3000/payment/success and localhost:3000/payment/failed

Reply:
This is to be modified when we have a page on FE for handling this. For now I'd probably should just set it to localhost:3000 to make sure it redirects to our FE page